docs/additional-functionality/delta-lake-support.md

```diff
@@ -54,8 +54,13 @@ GPU accelerated. These operations will fallback to the CPU.
 
 Delta Lake on Databricks has
 [automatic optimization](https://docs.databricks.com/optimizations/auto-optimize.html)
-features for optimized writes and automatic compaction.  Automatic compaction is not supported,
-and writes configured for automatic compaction will fallback to the CPU.
+features for optimized writes and automatic compaction.  Automatic compaction is supported
+when writing to directories via Scala/Python APIs, as follows:
+```scala
+my_dataframe.write.format("delta").mode("append").save("/path/to/delta-dir")
+```
+Automatic compaction is not supported to Delta tables via SQL. Auto compaction for such tables
+will automatically fall back to the CPU.
```

Comment:
This section confuses me.  What is it about automatic compaction that makes it only work with SQL writes?  Unless I'm missing something, automatic compaction is essentially tied to the write.  If the write is on the GPU, the automatic compaction associated with that write, whether that write originates from a simple Scala API, a SQL UPDATE or MERGE command, etc.  If the write is not on the GPU, then the automatic compaction will also not be on the GPU.  I'm not sure this distinction is worth calling out -- the query will still do the auto compaction in either case.  It's all about whether the write is on the GPU, and users will already think about optimized write/auto compact in terms of the write.

Reply:
The automatic compaction is tied to the write, because it's the write that is triggering the auto-compact and controlling how it runs.  If we don't hook into the transaction's writeFiles method to do a GPU write, we also won't hook into the post-commit hook added by that method to do a GPU auto-compact rather than a CPU auto-compact.  That's why it's tied to the GPU write.

Reply:
Yes, that makes perfect sense. Also, I've verified that SQL appends do not write via GPU, let alone compact.
I'll change the docs in a moment.

Reply:
I have removed that confusing section of the auto-compaction behaviour.